lib/to_do/api.ex

```diff
@@ -2,16 +2,34 @@ defmodule ToDo.API do
   @file_path Application.compile_env(:http_server, :file_path, "lib/to_do/data/to_dos.json")
 
   def create({:ok, data}) do
-    new_save_data =
-      File.read!(@file_path)
-      |> JSON.decode!()
-      |> Map.merge(data)
-      |> JSON.encode!()
-
-    File.write(@file_path, new_save_data, [:read, :write])
+    read_file()
+    |> add_new_todo(data)
+    |> write_file()
   end
 
   def create({:error, error_message}) do
     error_message
   end
+
+  defp read_file, do: File.read!(@file_path) |> JSON.decode!()
+
+  defp write_file(data), do: File.write(@file_path, JSON.encode!(data), [:read, :write])
```

Comment:
I know the implementation is using `File` but will that always be the case? what happens if the implementation changes and these functions would need to be renamed. What's a better name? and do these functions belong in their own abstraction?

Reply:
Better names could be `find_all`, `fetch_all`, or even just `all` instead of `read_file`, and perhaps `save` instead of `write_file`? 

As to whether these belong in their own abstraction, I'm sort of torn. In the past, I've handled these types of responsibilities using a db wrapper or query generator, and I guess in some ways that's what I'm creating here. I could move these functions to their own module called ToDo.DB or ToDo.Data so that what's being called is `Data.all/0` or `Data.save/1`

Reply:
I think a DB abstraction is a good idea.  `Data` is also a keyword / over used name, so be careful with it if that's what you choose.  Would be better to not alias it, and instead call `ToDo.Data` to make it explicit.

Reply:
think about what the domain language would be when choosing names, and avoid exposing the implementation in names as well, are generally good rules to follow.